ipn/ipnlocal/state_test.go

```diff
@@ -312,32 +312,30 @@ func TestStateMachine(t *testing.T) {
 
 	// Check that it hasn't called us right away.
 	// The state machine should be idle until we call Start().
-	assert.Equal(cc.getCalls(), []string{})
+	c.Assert(cc.getCalls(), qt.HasLen, 0)
```

Comment:
For what it's worth, I was trying to write all the cc.getCalls() assertions the same way. This makes it easier to move around the expected set of calls when later changing the state machine, without switching tests back and forth between HasLen-style and Equals-style.

Reply:
Ack. I switched because I was worried that using reflect.DeepEqual would end up distinguishing unnecessarily between `nil` and `[]string{}`, and that that would end up being brittle in an annoying way. (I don't really understand why we're returning `[]string{}` instead of nil.) Let's fix all of that in a follow-up.